feat: ability to get user details in Confluence toolsFeature/get user details

bijujoseph opened the pull request:
<!-- Thank you for your contribution! Please provide a brief summary. -->
Enables retrieval of user details referenced in pages (e.g., tagged users or entries in custom data forms like ConfiForms etc), which can then be leveraged in AI-driven workflows.

## Description

<!-- What does this PR do? Why is it needed? -->
The current codebase provides functions to fetch user details by accountId or username, but these are not exposed for use in automation. This PR introduces a new Confluence tool, get_user_details, which supports retrieving user information by accountId, username, or userkey.

This enhancement makes it easier for AI-driven workflows and integrations to access user details referenced in Confluence pages (e.g., tagged users or custom data forms like ConfiForms), enabling more seamless automation in Data Center and Server environments.

<!-- Link related issues: Fixes #<issue_number> -->

Fixes: #22

## Changes

<!-- Briefly list the key changes made. -->
- Added a new Confluence tool `get_user_details` to fetch user details by accountId, username, or userkey.
- Added a new function to `UserMixin` named (`get_user_details` and `get_user_details_by_userkey`)
- Exposed existing functions from `UserMixin` (`get_user_details_by_accountid`, `get_user_details_by_username`)
- Added adequate test cases for the changes made
- Updated the README 

## Testing

<!-- How did you test these changes? (e.g., unit tests, integration tests, manual checks) -->

- [x] Unit tests added/updated
- [x] Integration tests passed (**6 tests around jira epic failed** in `tests/test_real_api_validation.py`, everything else passed)
- [x] Manual checks performed: `verified the tool calls by building and deploying mcp server locally from my branch with changes`

**NOTE**: - the changes here are not associated with JIRA so I am ignoring those 6 failed ones relating to JIRA
## Checklist

- [x] Code follows project style guidelines (linting passes).
- [x] Tests added/updated for changes.
- [x] All tests pass locally. (except those 6)
- [x] Documentation updated (if needed).

bijujoseph commented:
@SharkyND  for your review. I have migrated this from https://github.com/sooperset/mcp-atlassian/pull/655

SharkyND commented:
Hey @bijujoseph thanks for PR. Can you please squash your commits and use convetional commits (https://www.conventionalcommits.org/en/v1.0.0/)?

bijujoseph committed:
feat(confluence): add ability to get user details

- Add user details functionality to Confluence tools
- Implement ConfluenceUser model in common.py
- Add comprehensive tests for user details feature
- Update server to expose user details endpoints

Resolves: #654
